core/src/main/java/com/linecorp/armeria/server/RouteCache.java

```diff
@@ -128,7 +128,7 @@ public Routed<V> find(RoutingContext routingCtx) {
             }
 
             final Routed<V> result = delegate.find(routingCtx);
-            if (result.isPresent() && !ambiguousRoutes.contains(result.route())) {
+            if (result.isPresent() && !dynamicPredicateRoutes.contains(result.route())) {
```

Comment:
Could we just use `isCacheable()`?
```suggestion
            if (result.isPresent() && result.route().isCacheable()) {
```

Reply:
The worry I had was that it is possible that a cache key is the same between a cacheable route and a non-cacheable route.

> - Extract all routes that have the same "routing key" as routes with dynamic predicates. This condition is needed because we can't differentiate from `routingCtx` between routes with and without dynamic predicates.
>    - i.e. `/routeA` with `headerPredicate` vs. just `/routeA`

Let me know if you think this is not an issue though 😅

Reply:
Pushed your suggestion - if I remember the test suite failed, but I guess it's better to verify via CI

Reply:
Verified via CI run: https://github.com/line/armeria/actions/runs/5879154437?pr=4934

Reply:
I understood. 
Minor suggestion) Could we first check `isCachable()` and then check `dynamicPredicateRoutes` because `isCachable()` is cheaper? 
```
if (result.isPresent() && result.route().isCacheable() &&
    !dynamicPredicateRoutes.contains(result.route())) {
```